packages/editor-ui/src/types/workflowHistory.ts

```diff
@@ -9,9 +9,10 @@ export type WorkflowHistory = {
 export type WorkflowVersionId = WorkflowHistory['versionId'];
 
 export type WorkflowVersion = WorkflowHistory & {
+	updatedAt: string;
 	nodes: IWorkflowDb['nodes'];
-	connection: IWorkflowDb['connections'];
-	workflow: IWorkflowDb;
+	connections: IWorkflowDb['connections'];
+	workflowId: IWorkflowDb['id'];
```

Comment:
We can also simplify with `Pick`.

Reply:
Thanks, it's a good Idea but I would still need to rename `id` to `workflowId` (that's how the backend sends)

So `WorkflowVersion` would look like this:
```ts
export type WorkflowVersion = WorkflowHistory & Pick<IWorkflowDb, 'nodes' | 'connections'> & {workflowId: IWorkflowDb['id']};
```

Or using a not very simple type utility that allows you to conditionally rename picked props
It would look something like this:

```ts
export type WorkflowVersion = WorkflowHistory & PickAndRename<IWorkflowDb, ['id' | 'nodes' | 'connections', {id: 'workflowId'}]>;
```

(_I already made `PickAndRename` but it feels too much and maybe we don't want to do this too many times to have this utility type in the project_ ;))

What do you think @ivov ?

Reply:
Good point, updated, please approve again